sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala

```diff
@@ -361,4 +352,84 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       assert(spark.sharedState.cacheManager.lookupCachedData(samePlan).isDefined)
     }
   }
-}
+
+  test("SPARK-27248 refreshTable should recreate cache with same cache name and storage level") {
+
+    // This section tests when a table is cached with its qualified name but its is refreshed with
+    // its unqualified name.
+    withTempDatabase { db =>
+      withTempView("cachedTable") {
+        withCache(s"$db.cachedTable") {
+
+          // Create table 'cachedTable' in default db for testing purpose.
+          sql(s"CREATE TABLE $db.cachedTable STORED AS PARQUET AS SELECT 1")
```

Comment:
We should not use `STORED AS PARQUET`. Otherwise, this table will be automatically converted to data source tables by `spark.sql.hive.convertMetastoreParquet=true (default)` and `spark.sql.hive.convertMetastoreCtas=true (default)`.